### ISSUE TITLE
Create a separate repo for examples

### ISSUE THREAD

**Issue by jsteinich** [hashicorp/terraform-cdk#749]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Create another repository (perhaps `terraform-cdk-examples`) to host the examples. This is something that has come up a couple times, but I don't believe formally tracked. Would allow us to remove the references to locally built distribution and replace with more standard user experience. May want to also expand readmes to better explain examples / setup.

### References

#745 
#551

**Comment by skorfmann** [comment]
> We're currently using the examples as some kind of smoke test ([.github/workflows/examples.yml](https://github.com/hashicorp/terraform-cdk/blob/main/.github/workflows/examples.yml#L38))
> It would be nice to keep something like that in place, when we move the examples to another repository.

Yes, that's the primary reason they are still part of this repo. In addition, I also find it quite useful for local manual testing without having to use links or something like yalc.

**Comment by jsteinich** [comment]
> We're currently using the examples as some kind of smoke test ([.github/workflows/examples.yml](https://github.com/hashicorp/terraform-cdk/blob/main/.github/workflows/examples.yml#L38))
> It would be nice to keep something like that in place, when we move the examples to another repository.

Yes, I meant to reference #558 since we could do event more testing.



> However, I don't quite understand, why we can't currently expand the readmes of the examples. That should also be possible while within this repo?

Absolutely. I just see moving to a different repo as a trigger to "let's improve the examples", but that could be done at anytime. Perhaps just opening this issue is enough of a starter.

I actually don't have a strong opinion on where they should be. I know there has been some informal discussion in the past and just wanted to continue that.
